app/src/lib/parser_txdef.h

```diff
@@ -225,18 +225,95 @@ typedef struct {
     uint32_t admin_th;
 } parser_updatemultisigmsg_t;
 
+#define PBIDX_UPDATEELECTORATEMSG_METADATA      1
+#define PBIDX_UPDATEELECTORATEMSG_ELECTORATE_ID 2
+#define PBIDX_UPDATEELECTORATEMSG_ELECTOR       3
 
-#define PBIDX_TX_FEES           1
-#define PBIDX_TX_MULTISIG       4
-#define PBIDX_TX_SENDMSG        51
-#define PBIDX_TX_UPDATEMSG      57
-#define PBIDX_TX_VOTEMSG        75
+#define PBIDX_UPDATEELECTORATEMSG_ELECTOR_MAX  8
+
+typedef struct {
+    struct {
+        unsigned int metadata : 1;
+        unsigned int electorate_id :1;
+    } seen;
+
+    const uint8_t *metadataPtr;
+    uint16_t metadataLen;
+    parser_metadata_t metadata;
+
+    const uint8_t *electorateIdPtr;
+    uint16_t electorateIdLen;
+
+    //Elector is a repeated field
+    uint8_t electorCount; //Total Electors fields in Tx
+    parser_participant_t elector_array[PBIDX_UPDATEELECTORATEMSG_ELECTOR_MAX];
+} parser_updateelectorate_t;
+
+#define PBIDX_CREATEPROPOSALMSG_METADATA    1
+#define PBIDX_CREATEPROPOSALMSG_TITLE       2
+#define PBIDX_CREATEPROPOSALMSG_OPTION      3
+#define PBIDX_CREATEPROPOSALMSG_DESCRIPTION 4
+#define PBIDX_CREATEPROPOSALMSG_RULEID      5
+#define PBIDX_CREATEPROPOSALMSG_STARTTIME   6
+#define PBIDX_CREATEPROPOSALMSG_AUTHOR      7
+
+typedef struct {
+    // These bits are to avoid duplicated fields
+    struct {
+        unsigned int metadata : 1;
+        unsigned int title : 1;
+        unsigned int raw_option : 1;
+        unsigned int description : 1;
+        unsigned int election_rule_id : 1;
+        unsigned int start_time: 1;
+        unsigned int author : 1;
+    } seen;
+
+    const uint8_t *metadataPtr;
+    uint16_t metadataLen;
+    parser_metadata_t metadata;
+
+    const uint8_t *titlePtr;
+    uint16_t titleLen;
+
+    const uint8_t *rawOptionPtr;
+    uint16_t rawOptionLen;
+
+    //UpdateElectorateMsg is encoded in raw_option field
+    const uint8_t *updateelectoratemsgPtr;
+    uint16_t updateelectoratemsgLen;
+    parser_updateelectorate_t updateelectoratemsg;
+
+    const uint8_t *descriptionPtr;
+    uint16_t descriptionLen;
+
+    const uint8_t *electionRuleIdPtr;
+    uint16_t electionRuleIdLen;
+
+    int64_t startTime;
+
+    const uint8_t *authorPtr;
+    uint16_t authorLen;
+} parser_createproposalmsg_t;
+
+
+
+
+#define PBIDX_TX_FEES                  1
+#define PBIDX_TX_MULTISIG              4
+#define PBIDX_TX_SENDMSG              51
+#define PBIDX_TX_UPDATE_MULTISIGMSG   57
+#define PBIDX_TX_CREATEPROPOSALMSG    73
+#define PBIDX_TX_VOTEMSG              75
+#define PBIDX_TX_UPDATEELECTORATEMSG  77
 
 typedef enum {
     Msg_Invalid = 0,
     Msg_Send,
```

Comment:
No need to change this unless it is quick.. but maybe we can improve a bit and unify the defines with the enum? something like:
```
typedef enum {
...
Msg_Vote = 75
```
we may need to be careful with the value for invalid type. 
We can discuss this later.. let's not do this now.

Reply:
ok!. Yes the invalid one should be an index that we're sure that is not reserve or anything on the proto messages like that (maybe a negative like -1?)

Reply:
-1 could be possible. Let's better make this change later. This PR is already big enough :)